client/core/core.go

```diff
@@ -1057,6 +1192,57 @@ func (c *Core) Trade(pw string, form *TradeForm) (*Order, error) {
 	return coreOrder, nil
 }
 
+// walletSet is a pair of wallets with asset configurations identified in useful
+// ways.
+type walletSet struct {
+	baseAsset  *dex.Asset
+	quoteAsset *dex.Asset
+	fromWallet *xcWallet
+	fromAsset  *dex.Asset
+	toWallet   *xcWallet
+	toAsset    *dex.Asset
+}
+
+// walletSet constructs a walletSet.
+func (c *Core) walletSet(dc *dexConnection, baseID, quoteID uint32, sell bool) (*walletSet, error) {
+	baseAsset, found := dc.assets[baseID]
+	if !found {
+		return nil, fmt.Errorf("unknown base asset %d -> %s for %s", baseID, unbip(baseID), dc.acct.url)
+	}
+	quoteAsset, found := dc.assets[quoteID]
+	if !found {
+		return nil, fmt.Errorf("unknown quote asset %d -> %s for %s", quoteID, unbip(quoteID), dc.acct.url)
+	}
+
+	// Connect and open the wallets if needed.
+	baseWallet, err := c.connectedWallet(baseID)
+	if err != nil {
+		return nil, err
+	}
+	quoteWallet, err := c.connectedWallet(quoteID)
+	if err != nil {
+		return nil, err
```

Comment:
If baseWallet is connected here, do you need to `baseWallet.connector.Disconnect()` or something like that?

Reply:
Not really. Once connected, a wallet should really never be disconnected unless directed by the user or at shutdown. "Locking" the wallet is different though, and should probably be done automatically when there are no active orders or matches. 

It might be a little convoluted here, because there are some assumptions made on the status of the wallet when this function is used. While they are for sure connected when returned by `walletSet`, they are not certain to be unlocked. Locked status is checked in `(*Core).Trade`, but maybe it should be checked here.

Reply:
Noting here that for the time being, and probably until a proper resolution to #214 is implemented, I'm not checking unlocked wallets here, but I did add a check for locked status in `isRedeemable` and `isSwappable` as a last check.

Reply:
I think 3f60788 fixes should clarify. Let me know if it's still not clear.